expo/expo
# [docs] Add redesigned table of contents

# Why

Part of [ENG-3963](https://linear.app/expo/issue/ENG-3963/simplify-the-table-of-contents-scroll-detection), [ENG-3962](https://linear.app/expo/issue/ENG-3962/add-new-components-as-separate-prs-in-expoexpo), Follow up of PR #16050

This adds the redesigned table of contents components. The designs didn't add special code block designs for headings containing code, like [the Notifications page](https://docs.expo.dev/versions/v44.0.0/sdk/notifications/#addnotificationreceivedlistenerlistener-event-notification--void-void).

That's why this TOC tries to resolve function names and just renders them as plain text links.

# How

page | light | dark
--- | --- | ---
**/get-started/installation** | ![image](https://user-images.githubusercontent.com/1203991/152789182-f502cbe7-4c84-40b4-8855-b919bd7314d0.png) | ![image](https://user-images.githubusercontent.com/1203991/152789157-dac024d3-51d0-45c5-8844-a46f2fd1492e.png)
**/tutorial/image** |  ![image](https://user-images.githubusercontent.com/1203991/152789248-44455bbf-7e59-46ab-89d1-16f8f1607b1e.png) | ![image](https://user-images.githubusercontent.com/1203991/152789267-49d18d4a-9359-43f6-83d8-f11c35dd6210.png)
**/versions/latest/sdk/notifications** | ![image](https://user-images.githubusercontent.com/1203991/152789021-00534c51-fa42-4b08-91d5-7ffebfad3839.png) | ![image](https://user-images.githubusercontent.com/1203991/152789079-743787b8-deea-4d4a-b3ca-f824c820a8b8.png)

# Test Plan

- Open various pages
- See how the TOC looks

# Checklist
